stacker/dag/__init__.py

```diff
@@ -161,6 +161,30 @@ def walk(self, walk_func):
         for n in nodes:
             walk_func(n)
 
+    def transitive_reduction(self):
```

Comment:
Note: This algorithm is currently way too un-performant to actually merge. I need to update this with a more efficient transitive reduction.